### PR TITLE
test: [M3-10365] - LKE-E "postLa" feature flag smoke tests

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝

This adds a handful of smoke tests that to confirm the state of the LKE cluster create and details pages when the LKE-E "postLa" feature flag is and is not enabled. It additionally adds some tests to confirm the pages work as expected when both "postLa" and "phase2Mtc" are enabled.

## Changes  🔄

- Add LKE-E "postLa" feature flag smoke tests

### Scope 🚢

 Upon production release, changes in this PR will be visible to:

- [ ] All customers
- [ ] Some customers (e.g. in Beta or Limited Availability)
- [x] No customers / Not applicable

## Target release date 🗓️

N/A

## How to test 🧪

We can rely on CI to confirm that the tests pass as expected. However, you can run the tests locally with this command:

```bash
pnpm cy:run -s "cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts"
```

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All tests and CI checks are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

### CODE REVIEW

**Comment by jdamore-linode** [comment]
Just noticed the merge conflict! I'll tackle that tomorrow, but removed the "Ready for Review" label in the meantime

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts [hunk 1]
```diff
@@ -67,146 +77,311 @@ describe('LKE-E Cluster Create', () => {
...
-    // Confirms LKE-E Phase 2 IP Stack and VPC options display with the flag ON.
-    cy.findByText('IP Stack').should('be.visible');
-    cy.findByText('IPv4', { exact: true }).should('be.visible');
-    cy.findByText('IPv4 + IPv6 (dual-stack)').should('be.visible');
-    cy.findByText('Automatically generate a VPC for this cluster').should(
-      'be.visible'
-    );
-    cy.findByText('Use an existing VPC').should('be.visible');
+  describe('Phase 2 MTC feature flag', () => {
+    it('Simple Page Check - Phase 2 MTC Flag ON', () => {
```
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:183]: I'm happy to help if the merge conflicts are annoying/complicated. I know that the lkeEnterprise2 flag change is going to be the cause of this.

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts [hunk 2]
```diff
@@ -67,146 +77,311 @@ describe('LKE-E Cluster Create', () => {
...
+      });
+    });
 
-    ui.regionSelect.find().click().type(`${mockRegions[0].label}`);
-    ui.regionSelect.findItemByRegionId(mockRegions[0].id).click();
+    /*
+     * - Confirms the state of the LKE create page when the LKE-E "postLa" flag is disabled.
+     * - Confirms that node pools are added directly via the plan table.
+     */
+    it('Simple Page Check - Post LA Flag OFF', () => {
```
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:145]: @jdamore-linode We could leave the "OFF" state test coverage for completeness, but since the postLa flag is enabled in prod now, and we haven't seen any need to retract that, it can be expected to remain on, and I'm not sure that the test is too useful. Do you think this test/any "Post LA OFF" is worth keeping?

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts (hunk 1 above)
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:183]: Thanks @mjac0bs! I think I should be good, but I'll reach out if I run into any trouble 👍 I didn't get a chance to tackle this yesterday like I said I would, but I'm working on it right now!

File: packages/manager/cypress/e2e/core/cloudpulse/create-user-alert.spec.ts
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/cloudpulse/create-user-alert.spec.ts:1]: I think all these random CloudPulse changes are a result of Prettier running automatically when committing the merge from `develop` into this branch; I don't think the CloudPulse team runs Prettier when they commit, so these all got modified to apply our Prettier rules when I made my commit

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts (hunk 1 above)
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:183]: Just pushed that @mjac0bs! If you wouldn't mind taking a quick look to make sure I didn't accidentally delete or mess up any of the tests you added in #12826 I'd appreciate it!

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts [hunk 3]
```diff
@@ -48,17 +62,13 @@ const mockVPC = vpcFactory.build({
...
-];
+const mockRegion = regionFactory.build({
+  capabilities: ['Linodes', 'Kubernetes', 'Kubernetes Enterprise', 'VPCs'],
+});
 
 /**
- * - Confirms VPC and IP Stack selections are shown with the respective `phase2Mtc` feature flags enabled.
- * - Confirms VPC and IP Stack selections are not shown in create flow with their respective `phase2Mtc` feature flags disabled.
+ * - Confirms VPC and IP Stack selections are shown with `phase2Mtc` feature flag is enabled.
+ * - Confirms VPC and IP Stack selections are not shown in create flow with `phase2Mtc` feature flag is disabled.
```
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:71]: Minor regression from the merge. The deleted text alludes to multiple (nested) phase2 feature flags, rather than just a bool.

Though I think we should move this comment to the [section](https://github.com/linode/manager/pull/12886/files?diff=unified#diff-03cbaa147b1d7544e682480406a247071f4332e5867c112f35bd4875f59c4201R186) that tests the phase 2 flows.

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts (hunk 2 above)
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:145]: @mjac0bs I'd prefer to have the off state coverage if it's not too much of a maintenance burden! I think you're right and can't really imagine any scenarios where we'd have to turn the flag off that aren't farfetched, but on the other hand if we could anticipate every edge case scenario we wouldn't really need tests in the first place 😅 I'd really hate to be in a situation where we need to disable a flag on short notice only to find out that the off state is broken

**Review by mjac0bs — APPROVED** [review]
Thank you for this and handling the combinatorial explosion of feature flag cases in this file. 😅 

I think your merge preserved the existing test cases from my previous PR, and thank you for the additions.

File: packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts (hunk 3 above)
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/core/kubernetes/smoke-lke-enterprise.spec.ts:71]: Nice catch, thanks @mjac0bs! Just improved that wording and moved it to be above the "Phase 2 MTC feature flag" describe block
